https://github.com/sektioneins/suhosin/commit/1a2cb68a84: header value leaked when
suhosin.upload.max_newlines is exceeded

diff --git suhosin-0.9.38.orig/rfc1867_new.c suhosin-0.9.38/rfc1867_new.c
index 362229e..8bb9f6e 100644
--- suhosin-0.9.38.orig/rfc1867_new.c
+++ suhosin-0.9.38/rfc1867_new.c
@@ -439,13 +439,14 @@ static int multipart_buffer_headers(multipart_buffer *self, zend_llist *header T
 			newlines = 0;
 
 		} else if (buf_value.c) { /* If no ':' on the line, add to previous line */
-			smart_str_appends(&buf_value, line);
 			newlines++;
 			if (newlines > SUHOSIN_G(upload_max_newlines)) {
 				SUHOSIN_G(abort_request) = 1;
 				suhosin_log(S_FILES, "configured maximum number of newlines in RFC1867 MIME headers limit exceeded - dropping rest of upload");
+				smart_str_free(&buf_value);
 				return 0;
 			}
+			smart_str_appends(&buf_value, line);
 
 		} else {
 			continue;
